en/guide/define_xml_element.md

```diff
@@ -222,16 +225,42 @@ Care should still be taken to ensure that any changes that alter the way that th
 Messages are very rarely deleted, as this may break compatibility with legacy MAVLink 1 hardware that is unlikely to be updated to more recent versions.
 
 
-#### MAVLink 2 Message Extensions {#message_extensions}
+### Message Extensions (MAVLink 2) {#message_extensions}
 
-The exception to the above rule is that when using [MAVLink 2](../guide/mavlink_2.md) you can add new fields to messages with an `id` of 0 - 255 using the [<extensions>](../guide/mavlink_2.md#message_extensions) tag *without breaking compatibility*. 
-These fields will not be sent when the MAVLink 1 protocol is used.
+MAVLink 2 defines *extension fields*, which can be added to an existing message without breaking binary compatibility for receivers that have not been updated.
 
-Otherwise the rules are the same; once added you cannot modify or remove fields. 
-You can however continue to add new fields as long as you do not exceed the maximum field number or payload size limits.
+<!-- add note here WHY you would use this:  -->
```

Comment:
@LorenzMeier My assumption is that we use extensions because it is hard to coordinate message changes for existing systems (something we do want to improve going forward). 
I presume the reason we don't recommend these everywhere (ie drop the idea of reordering) is that there is benefit in reordering and in having the CRCEXTRA as a compatibility check?

Reply:
This is now added as a [general question](https://github.com/mavlink/mavlink-devguide/issues/83#issue-342160313). I'm merging this update as-is.